### PR TITLE
Add detection eval optimizations.

### PR DESCRIPTION
**Pull request by benjaminrwilson**
Speeds up the multiprocessing in the detection eval + transitions classes to `dataclasses`

### CODE REVIEW

File: argoverse/evaluation/detection/eval.py [hunk 1]
```diff
@@ -119,29 +117,29 @@ def evaluate(self) -> pd.DataFrame:
         gt_fpaths = list(self.gt_root_fpath.glob("*/per_sweep_annotations_amodal/*.json"))
 
         assert len(dt_fpaths) == len(gt_fpaths)
-        data: DefaultDict[str, np.ndarray] = defaultdict(list)
+        data: DefaultDict[str, List[np.ndarray]] = defaultdict(list)
         cls_to_ninst: DefaultDict[str, int] = defaultdict(int)
 
+        jobs = [AccumulateJob(self.dt_root_fpath, gt_fpath, self.cfg, self.avm) for gt_fpath in gt_fpaths]
+        self.num_procs = 1
```
Comment (tagarwal-argoai) [inline argoverse/evaluation/detection/eval.py:124]: Why is this being set here to 1?

File: argoverse/evaluation/detection/utils.py [hunk 2]
```diff
@@ -99,15 +100,28 @@ class DetectionCfg(NamedTuple):
...
+        avm: Argoverse map object.
+    """
+
+    dt_root_fpath: Path
+    gt_fpath: Path
+    cfg: DetectionCfg
+    avm: Optional[ArgoverseMap]
+
+
+def accumulate(job: AccumulateJob) -> Tuple[DefaultDict[str, np.ndarray], DefaultDict[str, int]]:
```
Comment (tagarwal-argoai) [inline argoverse/evaluation/detection/utils.py:120]: Can this not be a instance method of AccumulateJob?

**Review by tagarwal-argoai — COMMENTED** [review]
Couple of comments here as well.

File: argoverse/evaluation/detection/eval.py (hunk 1 above)
Comment (benjaminrwilson) [reply argoverse/evaluation/detection/eval.py]: Whoops, thanks for catching that.

File: argoverse/evaluation/detection/utils.py (hunk 2 above)
Comment (benjaminrwilson) [inline argoverse/evaluation/detection/utils.py:120]: What do you think the best way for us to pass this into a mapping function would be (e.g., `ProcessMap`?

**Comment by johnwlambert** [comment]
Thanks for the PR, @benjaminrwilson. It looks like CI is failing:
```
FAILED tests/test_eval_detection.py::test_accumulate - TypeError: accumulate(...
FAILED tests/test_eval_detection.py::test_remove_duplicate_instances_ground_truth
FAILED tests/test_eval_detection.py::test_AP_on_filtered_instances - ValueErr...
```
Do you mind taking a look?

File: argoverse/evaluation/detection/utils.py (hunk 2 above)
Comment (tagarwal-argoai) [reply argoverse/evaluation/detection/utils.py:120]: Discussed offline on slack. I think this is fine!

Comment (tagarwal-argoai) [reply argoverse/evaluation/detection/utils.py:120]: Not relevant anymore.

**Review by tagarwal-argoai — APPROVED** [review]
LGTM!

**Pull request merged**